### ISSUE TITLE
Broken pipe with large input

### ISSUE THREAD

**Issue by mcastorina** [mcastorina/heavi#1]
```
» seq -w 1 5000 | heavi 0100 | heavi -v 4321 > /dev/null
Error: IOError(Broken pipe (os error 32))
```

**Comment by mcastorina** [comment]
Strangely enough, this only happens for release builds

**Comment by mcastorina** [comment]
Fixed in 4d49f16a08f94f38b7091912f34d4e8698740d74

The problem was assuming there was no more data to read when we didn't fill up the buffer entirely.
